apps/vercel/src/clients/Vercel.ts

```diff
@@ -0,0 +1,76 @@
+import { CreateDeploymentInput, Deployment, ListProjectsResponse } from '../types';
+
+interface VercelAPIClient {
+  checkToken: () => Promise<boolean>;
+  listProjects: () => Promise<ListProjectsResponse>;
+  createDeployment: (input: CreateDeploymentInput) => Promise<Deployment>;
+  getDeploymentById: (deploymentId: string) => Promise<Deployment>;
+}
+
+export default class VercelClient implements VercelAPIClient {
+  baseEndpoint: string;
+  accessToken: string;
+
+  constructor(accessToken: string = '') {
```

Comment:
I've recently started loving the TS shorthand for this which gets rid of boilerplate:

```
  constructor(
    public baseEndpoint: string = 'https://api.vercel.com',
    public accessToken: string = ''
  ) {}
```

See https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd